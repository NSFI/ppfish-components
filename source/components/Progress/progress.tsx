import * as React from 'react';
import Icon from '../Icon';
import { Circle } from './src/index.js';
import classNames from 'classnames';

const statusColorMap = {
  normal: '#108ee9',
  exception: '#ff5500',
  success: '#87d068',
};

export type ProgressType = 'line' | 'circle' | 'dashboard';
export type ProgressSize = 'default' | 'small';

export interface ProgressProps {
  prefixCls?: string;
  className?: string;
  message?: string;
  extraContent?: React.ReactNode | null;
  operation?: React.ReactNode | null;
  type?: ProgressType;
  percent?: number;
  successPercent?: number;
  format?: (percent?: number, successPercent?: number) => string;
  status?: 'success' | 'active' | 'exception';
  showInfo?: boolean;
  strokeWidth?: number;
  strokeLinecap?: string;
  strokeColor?: string;
  trailColor?: string;
  width?: number;
  style?: React.CSSProperties;
  gapDegree?: number;
  gapPosition?: 'top' | 'bottom' | 'left' | 'right';
  size?: ProgressSize;
}

const validProgress = (progress: number | undefined) => {
  if (!progress || progress < 0) {
    return 0;
  } else if (progress > 100) {
    return 100;
  }
  return progress;
};

export default class Progress extends React.Component<ProgressProps, {}> {
  static Line: any;

  static Circle: any;

  static defaultProps = {
    type: 'line' as ProgressType,
    percent: 0,
    showInfo: true,
    trailColor: '#f3f3f3',
    prefixCls: 'fishd-progress',
    size: 'default' as ProgressSize,
  };

  render() {
    const props = this.props;
    const {
      prefixCls,
      className,
      percent = 0,
      status,
      format,
      trailColor,
      size,
      successPercent,
      type,
      strokeWidth,
      width,
      showInfo,
      gapDegree = 0,
      gapPosition,
      strokeColor,
      strokeLinecap = 'round',
      extraContent,
      message,
      operation,
      ...restProps
    } = props;
    const progressStatus =
      parseInt(successPercent ? successPercent.toString() : percent.toString(), 10) >= 100 &&
      !('status' in props)
        ? 'success'
        : status || 'normal';
    let progressInfo;
    let progress;
    const textFormatter = format || (percentNumber => `${percentNumber}%`);

    if (showInfo) {
      let text;
      const iconType = type === 'circle' || type === 'dashboard' ? 'hints-alone-' : 'hints-';
      if (format || (progressStatus !== 'exception' && progressStatus !== 'success')) {
        text = textFormatter(validProgress(percent), validProgress(successPercent));
      } else if (progressStatus === 'exception') {
        text = <Icon type={`${iconType}error`} />;
      } else if (progressStatus === 'success') {
        text = <Icon type={`${iconType}success`} />;
      }
      progressInfo = <span className={`${prefixCls}-text`}>{text}</span>;
    }

    if (type === 'line') {
      const percentStyle = {
        width: `${validProgress(percent)}%`,
        height: strokeWidth || (size === 'small' ? 6 : 8),
        background: strokeColor,
        borderRadius: strokeLinecap === 'square' ? 0 : '100px',
      };
      const successPercentStyle = {
        width: `${validProgress(successPercent)}%`,
        height: strokeWidth || (size === 'small' ? 6 : 8),
        borderRadius: strokeLinecap === 'square' ? 0 : '100px',
      };
      const successSegment =
        successPercent !== undefined ? (
          <div className={`${prefixCls}-success-bg`} style={successPercentStyle} />
        ) : null;

      progress = (
        <div className={`${prefixCls}-line-ctner`}>
          <div className={`${prefixCls}-basic`}>
            <div className={`${prefixCls}-outer`}>
              <div className={`${prefixCls}-inner`}>
                <div className={`${prefixCls}-bg`} style={percentStyle}>
                  {extraContent ? <div className={`${prefixCls}-extra`}>{extraContent}</div> : null}
                </div>
                {successSegment}
              </div>
            </div>
            {progressInfo}
          </div>
          {message ? <span className={`${prefixCls}-msg`}>{message}</span> : null}
          {operation ? <span className={`${prefixCls}-oper`}>{operation}</span> : null}
        </div>
      );
    } else if (type === 'circle' || type === 'dashboard') {
      const circleSize = width || 120;
      const circleStyle = {
        width: circleSize,
        height: circleSize,
        fontSize: circleSize * 0.15 + 6,
      };
      const circleWidth = strokeWidth || 6;
      const gapPos = gapPosition || (type === 'dashboard' && 'bottom') || 'top';
      const gapDeg = gapDegree || (type === 'dashboard' && 75);
      progress = (
        <div className={`${prefixCls}-inner`} style={circleStyle}>
          <Circle
            percent={validProgress(percent)}
            strokeWidth={circleWidth}
            trailWidth={circleWidth}
            strokeColor={(statusColorMap as any)[progressStatus]}
            strokeLinecap={strokeLinecap}
            trailColor={trailColor}
            prefixCls={prefixCls}
            gapDegree={gapDeg}
            gapPosition={gapPos}
          />
          <div className={`${prefixCls}-circle-info`}>
            {progressInfo}
            {message ? <span className={`${prefixCls}-msg`}>{message}</span> : null}
          </div>
        </div>
      );
    }

    const classString = classNames(
      prefixCls,
      {
        [`${prefixCls}-${(type === 'dashboard' && 'circle') || type}`]: true,
        [`${prefixCls}-status-${progressStatus}`]: true,
        [`${prefixCls}-show-info`]: showInfo,
        [`${prefixCls}-${size}`]: size,
      },
      className,
    );

    return (
      <div {...restProps} className={classString}>
        {progress}
        {(type === 'circle' || type === 'dashboard') && operation ? (
          <span className={`${prefixCls}-oper`}>{operation}</span>
        ) : null}
      </div>
    );
  }
}
