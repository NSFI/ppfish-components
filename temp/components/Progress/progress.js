var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import PropTypes from 'prop-types';
import * as React from 'react';
import Icon from '../Icon';
import { Circle } from './src/index.js';
import classNames from 'classnames';
var statusColorMap = {
    normal: '#108ee9',
    exception: '#ff5500',
    success: '#87d068'
};
var validProgress = function (progress) {
    if (!progress || progress < 0) {
        return 0;
    }
    else if (progress > 100) {
        return 100;
    }
    return progress;
};
var Progress = /** @class */ (function (_super) {
    __extends(Progress, _super);
    function Progress() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Progress.prototype.render = function () {
        var _a;
        var props = this.props;
        var prefixCls = props.prefixCls, className = props.className, _b = props.percent, percent = _b === void 0 ? 0 : _b, status = props.status, format = props.format, trailColor = props.trailColor, size = props.size, successPercent = props.successPercent, type = props.type, strokeWidth = props.strokeWidth, width = props.width, showInfo = props.showInfo, _c = props.gapDegree, gapDegree = _c === void 0 ? 0 : _c, gapPosition = props.gapPosition, strokeColor = props.strokeColor, _d = props.strokeLinecap, strokeLinecap = _d === void 0 ? 'round' : _d, extraContent = props.extraContent, message = props.message, operation = props.operation, restProps = __rest(props, ["prefixCls", "className", "percent", "status", "format", "trailColor", "size", "successPercent", "type", "strokeWidth", "width", "showInfo", "gapDegree", "gapPosition", "strokeColor", "strokeLinecap", "extraContent", "message", "operation"]);
        var progressStatus = parseInt(successPercent ? successPercent.toString() : percent.toString(), 10) >= 100 &&
            !('status' in props)
            ? 'success'
            : status || 'normal';
        var progressInfo;
        var progress;
        var textFormatter = format || (function (percentNumber) { return percentNumber + "%"; });
        if (showInfo) {
            var text = void 0;
            var iconType = type === 'circle' || type === 'dashboard' ? 'hints-alone-' : 'hints-';
            if (format || (progressStatus !== 'exception' && progressStatus !== 'success')) {
                text = textFormatter(validProgress(percent), validProgress(successPercent));
            }
            else if (progressStatus === 'exception') {
                text = React.createElement(Icon, { type: iconType + "error" });
            }
            else if (progressStatus === 'success') {
                text = React.createElement(Icon, { type: iconType + "success" });
            }
            progressInfo = React.createElement("span", { className: prefixCls + "-text" }, text);
        }
        if (type === 'line') {
            var percentStyle = {
                width: validProgress(percent) + "%",
                height: strokeWidth || (size === 'small' ? 6 : 8),
                background: strokeColor,
                borderRadius: strokeLinecap === 'square' ? 0 : '100px'
            };
            var successPercentStyle = {
                width: validProgress(successPercent) + "%",
                height: strokeWidth || (size === 'small' ? 6 : 8),
                borderRadius: strokeLinecap === 'square' ? 0 : '100px'
            };
            var successSegment = successPercent !== undefined ? (React.createElement("div", { className: prefixCls + "-success-bg", style: successPercentStyle })) : null;
            progress = (React.createElement("div", { className: prefixCls + "-line-ctner" },
                React.createElement("div", { className: prefixCls + "-basic" },
                    React.createElement("div", { className: prefixCls + "-outer" },
                        React.createElement("div", { className: prefixCls + "-inner" },
                            React.createElement("div", { className: prefixCls + "-bg", style: percentStyle }, extraContent ? React.createElement("div", { className: prefixCls + "-extra" }, extraContent) : null),
                            successSegment)),
                    progressInfo),
                message ? React.createElement("span", { className: prefixCls + "-msg" }, message) : null,
                operation ? React.createElement("span", { className: prefixCls + "-oper" }, operation) : null));
        }
        else if (type === 'circle' || type === 'dashboard') {
            var circleSize = width || 120;
            var circleStyle = {
                width: circleSize,
                height: circleSize,
                fontSize: circleSize * 0.15 + 6
            };
            var circleWidth = strokeWidth || 6;
            var gapPos = gapPosition || (type === 'dashboard' && 'bottom') || 'top';
            var gapDeg = gapDegree || (type === 'dashboard' && 75);
            progress = (React.createElement("div", { className: prefixCls + "-inner", style: circleStyle },
                React.createElement(Circle, { percent: validProgress(percent), strokeWidth: circleWidth, trailWidth: circleWidth, strokeColor: statusColorMap[progressStatus], strokeLinecap: strokeLinecap, trailColor: trailColor, prefixCls: prefixCls, gapDegree: gapDeg, gapPosition: gapPos }),
                React.createElement("div", { className: prefixCls + "-circle-info" },
                    progressInfo,
                    message ? React.createElement("span", { className: prefixCls + "-msg" }, message) : null)));
        }
        var classString = classNames(prefixCls, (_a = {},
            _a[prefixCls + "-" + ((type === 'dashboard' && 'circle') || type)] = true,
            _a[prefixCls + "-status-" + progressStatus] = true,
            _a[prefixCls + "-show-info"] = showInfo,
            _a[prefixCls + "-" + size] = size,
            _a), className);
        return (React.createElement("div", __assign({}, restProps, { className: classString }),
            progress,
            (type === 'circle' || type === 'dashboard') && operation ? (React.createElement("span", { className: prefixCls + "-oper" }, operation)) : null));
    };
    Progress.defaultProps = {
        type: 'line',
        percent: 0,
        showInfo: true,
        trailColor: '#f3f3f3',
        prefixCls: 'fishd-progress',
        size: 'default'
    };
    Progress.propTypes = {
        status: PropTypes.oneOf(['normal', 'exception', 'active', 'success']),
        type: PropTypes.oneOf(['line', 'circle', 'dashboard']),
        showInfo: PropTypes.bool,
        percent: PropTypes.number,
        width: PropTypes.number,
        strokeWidth: PropTypes.number,
        strokeLinecap: PropTypes.oneOf(['round', 'square']),
        strokeColor: PropTypes.string,
        trailColor: PropTypes.string,
        format: PropTypes.func,
        gapDegree: PropTypes.number,
        default: PropTypes.oneOf(['default', 'small'])
    };
    return Progress;
}(React.Component));
export default Progress;
