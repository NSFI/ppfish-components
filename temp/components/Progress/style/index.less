@import (reference) "../../../assets/css/index";

@progress-prefix-cls: ~"@{fishd-prefix}-progress";

.@{progress-prefix-cls} {
  .reset-component;

  position: relative;
  display: inline-block;

  &-line {
    position: relative;
    width: 100%;
    font-size: @font-size-base;
  }

  &-small&-line,
  &-small&-line &-text .@{iconfont-prefix},
  &-small&-line &-msg .@{iconfont-prefix} {
    font-size: @font-size-sm;
  }

  &-line-ctner {
    display: inline-flex;
    width: 100%;
  }

  &-basic {
    flex: 1;
  }

  &-outer {
    display: inline-block;
    width: 100%;
    margin-right: 0;
    padding-right: 0;
    .@{progress-prefix-cls}-show-info & {
      padding-right: ~"calc(2em + 8px)";
      margin-right: ~"calc(-2em - 8px)";
    }
  }

  &-inner {
    display: inline-block;
    width: 100%;
    background-color: @progress-remaining-color;
    border-radius: 100px;
    vertical-align: middle;
    position: relative;
  }

  &-extra {
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 20px;
    height: 20px;
    font-size: 20px;
    right: -10px;
    top: 50%;
    transform: translateY(-50%);
    // border-radius: 50%;
    // overflow: hidden;
  }

  &-circle-info {
    display: flex;
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    margin: 0;
    flex-direction: column;
    justify-content: center;
    // font-size: @font-size-base;

    & > span {
      /* stylelint-disable-next-line declaration-no-important */
      position: relative !important;
      /* stylelint-disable-next-line declaration-no-important */
      top: 0 !important;
      /* stylelint-disable-next-line declaration-no-important */
      transform: none !important;
    }

    .@{progress-prefix-cls}-msg {
      font-size: @font-size-base;
    }
  }

  &-circle-trail {
    stroke: @progress-remaining-color;
  }

  &-circle-path {
    stroke: @progress-default-color;
    animation: ~"@{fishd-prefix}-progress-appear" 0.3s;
  }

  &-success-bg,
  &-bg {
    background-color: @progress-default-color;
    transition: all 0.4s @ease-out-circ 0s;
    position: relative;
  }

  &-success-bg {
    background-color: @success-color;
    position: absolute;
    top: 0;
    left: 0;
  }

  &-text {
    word-break: normal;
    width: 2em;
    text-align: left;
    font-size: 1em;
    margin-left: 8px;
    vertical-align: middle;
    display: inline-block;
    color: @text-color-secondary;
    line-height: 1;
    .@{iconfont-prefix} {
      font-size: @font-size-base;
    }
  }

  &-msg {
    font-size: 1em;
    margin-left: 8px;
    display: inline-block;
    color: @text-color-secondary;

    .@{iconfont-prefix} {
      font-size: @font-size-base;
    }
  }

  &-oper {
    font-size: 1em;
    margin-left: 8px;
    display: inline-block;
    // color: @primary-color;

    .@{iconfont-prefix} {
      font-size: @font-size-base;
    }
  }

  &-circle &-oper {
    display: flex;
    margin: 8px 0 0 0;
    justify-content: center;
  }

  &-status-active {
    .@{progress-prefix-cls}-bg:before {
      content: "";
      opacity: 0;
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      background: @component-background;
      border-radius: 10px;
      animation: ~"@{fishd-prefix}-progress-active" 2.4s @ease-out-quint infinite;
    }
  }

  &-status-exception {
    .@{progress-prefix-cls}-bg {
      background-color: @error-color;
    }
    .@{progress-prefix-cls}-text,
    .@{progress-prefix-cls}-msg {
      color: @error-color;
    }
    .@{progress-prefix-cls}-circle-path {
      stroke: @error-color;
    }
  }

  &-status-success {
    .@{progress-prefix-cls}-bg {
      background-color: @success-color;
    }
    .@{progress-prefix-cls}-text,
    .@{progress-prefix-cls}-msg {
      color: @success-color;
    }
    .@{progress-prefix-cls}-circle-path {
      stroke: @success-color;
    }
  }

  &-circle &-inner {
    position: relative;
    line-height: 1;
    background-color: transparent;
  }

  &-circle &-text,
  &-circle &-msg {
    display: block;
    position: absolute;
    width: 100%;
    text-align: center;
    line-height: 1;
    top: 50%;
    transform: translateY(-50%);
    left: 0;
    margin: 0;
    color: @text-color;

    .@{iconfont-prefix} {
      font-size: 14 / 12em;
    }
  }

  &-circle&-status-exception {
    .@{progress-prefix-cls}-text,
    .@{progress-prefix-cls}-msg {
      color: @error-color;
    }
  }
  &-circle&-status-success {
    .@{progress-prefix-cls}-text,
    .@{progress-prefix-cls}-msg {
      color: @success-color;
    }
  }
}

@keyframes ~"@{fishd-prefix}-progress-active" {
  0% {
    opacity: 0.1;
    width: 0;
  }

  20% {
    opacity: 0.5;
    width: 0;
  }

  100% {
    opacity: 0;
    width: 100%;
  }
}
