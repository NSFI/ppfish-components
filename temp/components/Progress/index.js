import Progress from './progress';
import './style/index.less';
export default Progress;
