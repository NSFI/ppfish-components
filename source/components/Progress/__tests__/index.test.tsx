import React from 'react';
import { mount } from 'enzyme';
import Progress from '../index';

describe('Progress', () => {
  it('successPercent should decide the progress status when it exists', () => {
    const wrapper = mount(<Progress percent={100} successPercent={50} />);
    expect(wrapper.find('.fishd-progress-status-success')).toHaveLength(0);

    wrapper.setProps({ percent: 50, successPercent: 100 });
    expect(wrapper.find('.fishd-progress-status-success')).toHaveLength(1);
  });

  it('render out-of-range progress', () => {
    const wrapper = mount(<Progress percent={120} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render out-of-range progress with info', () => {
    const wrapper = mount(<Progress percent={120} showInfo />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render negetive progress', () => {
    const wrapper = mount(<Progress percent={-20} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render negetive successPercent', () => {
    const wrapper = mount(<Progress percent={50} successPercent={-20} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render format', () => {
    const wrapper = mount(
      <Progress
        percent={50}
        successPercent={10}
        format={(percent, successPercent) => `${percent} ${successPercent}`}
      />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
