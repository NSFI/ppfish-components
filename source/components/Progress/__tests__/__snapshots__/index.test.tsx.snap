// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress render format 1`] = `
<div
  class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
>
  <div
    class="fishd-progress-line-ctner"
  >
    <div
      class="fishd-progress-basic"
    >
      <div
        class="fishd-progress-outer"
      >
        <div
          class="fishd-progress-inner"
        >
          <div
            class="fishd-progress-bg"
            style="width: 50%; height: 8px; border-radius: 100px;"
          />
          <div
            class="fishd-progress-success-bg"
            style="width: 10%; height: 8px; border-radius: 100px;"
          />
        </div>
      </div>
      <span
        class="fishd-progress-text"
      >
        50 10
      </span>
    </div>
  </div>
</div>
`;

exports[`Progress render negetive progress 1`] = `
<div
  class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
>
  <div
    class="fishd-progress-line-ctner"
  >
    <div
      class="fishd-progress-basic"
    >
      <div
        class="fishd-progress-outer"
      >
        <div
          class="fishd-progress-inner"
        >
          <div
            class="fishd-progress-bg"
            style="width: 0%; height: 8px; border-radius: 100px;"
          />
        </div>
      </div>
      <span
        class="fishd-progress-text"
      >
        0%
      </span>
    </div>
  </div>
</div>
`;

exports[`Progress render negetive successPercent 1`] = `
<div
  class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
>
  <div
    class="fishd-progress-line-ctner"
  >
    <div
      class="fishd-progress-basic"
    >
      <div
        class="fishd-progress-outer"
      >
        <div
          class="fishd-progress-inner"
        >
          <div
            class="fishd-progress-bg"
            style="width: 50%; height: 8px; border-radius: 100px;"
          />
          <div
            class="fishd-progress-success-bg"
            style="width: 0%; height: 8px; border-radius: 100px;"
          />
        </div>
      </div>
      <span
        class="fishd-progress-text"
      >
        50%
      </span>
    </div>
  </div>
</div>
`;

exports[`Progress render out-of-range progress 1`] = `
<div
  class="fishd-progress fishd-progress-line fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
>
  <div
    class="fishd-progress-line-ctner"
  >
    <div
      class="fishd-progress-basic"
    >
      <div
        class="fishd-progress-outer"
      >
        <div
          class="fishd-progress-inner"
        >
          <div
            class="fishd-progress-bg"
            style="width: 100%; height: 8px; border-radius: 100px;"
          />
        </div>
      </div>
      <span
        class="fishd-progress-text"
      >
        <i
          class="fishdicon fishdicon-hints-success"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`Progress render out-of-range progress with info 1`] = `
<div
  class="fishd-progress fishd-progress-line fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
>
  <div
    class="fishd-progress-line-ctner"
  >
    <div
      class="fishd-progress-basic"
    >
      <div
        class="fishd-progress-outer"
      >
        <div
          class="fishd-progress-inner"
        >
          <div
            class="fishd-progress-bg"
            style="width: 100%; height: 8px; border-radius: 100px;"
          />
        </div>
      </div>
      <span
        class="fishd-progress-text"
      >
        <i
          class="fishdicon fishdicon-hints-success"
        />
      </span>
    </div>
  </div>
</div>
`;
